pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -380,6 +381,16 @@ func (am *Alertmanager) buildReceiverIntegrations(receiver *apimodels.PostableAp
 	var integrations []notify.Integration
 
 	for i, r := range receiver.GrafanaManagedReceivers {
+		// secure settings are already encrypted at this point
+		secureSettings := securejsondata.SecureJsonData(make(map[string][]byte, len(r.SecureSettings)))
+
+		for k, v := range r.SecureSettings {
+			d, err := base64.StdEncoding.DecodeString(v)
+			if err != nil {
+				return nil, fmt.Errorf("failed to decode secure setting")
+			}
+			secureSettings[k] = d
+		}
```

Comment:
All channels use `model.DecryptedValue` explicitly where required. So we don't need to decrypt it here.

Reply:
https://github.com/grafana/grafana/blob/baf80dad82dad4ead05455123268bacda9554726/pkg/services/ngalert/notifier/channels/pagerduty.go#L52

Reply:
Here, we don't decrypt it; we just decoded from `base64`. These are two separate things: encryption that encrypts the secrets and its result is `[]byte` and conversion of this result to a `base64` string for storing it to the database as string.